scripts/validate_docstrings.py

```diff
@@ -490,10 +491,41 @@ def yields(self):
     @property
     def method_source(self):
         try:
-            return inspect.getsource(self.obj)
+            source = inspect.getsource(self.obj)
+            # Remove common indentation.
+            first_spaces = re.match(' +', source)
+            if first_spaces:
+                first_spaces = first_spaces.group(0)
+                source = re.sub('^' + first_spaces, '', source)
+                source = re.sub(r'\n' + first_spaces, r'\n', source)
```

Comment:
What is happening here is that, when `inspect.getsource()` runs, it returns the function's code as it is on the source file, indentation included. So, for example, if the function is a method:

```
class A:
    def B(self):
        return None
```

Getting the source of B will return
```
    def B(self):
        return None
```
with the previously present indentation kept.

If we feed this directly to `ast.parse()`, it causes an indentation error, since the first thing read on the string is a space, and it is unexpected.

An easy fix is to apply `.strip()` to the string containing the function source, which remove all whitespaces in the beginning and the end. In my example, the result would be:
```
def B(self):
        return None
```
Which solves the previous problem.

The thing is, on more complex cases, this approach isn't enough. Consider the real example:
```
    @Appender(_shared_docs['melt'] %
              dict(caller='df.melt(',
                   versionadded='.. versionadded:: 0.20.0\n',
                   other='melt'))
    def melt(self, id_vars=None, value_vars=None, var_name=None,
             value_name='value', col_level=None):
        from pandas.core.reshape.melt import melt
        return melt(self, id_vars=id_vars, value_vars=value_vars,
                    var_name=var_name, value_name=value_name,
                    col_level=col_level)
```
By applying `.strip()`, we get:
```
@Appender(_shared_docs['melt'] %
              dict(caller='df.melt(',
                   versionadded='.. versionadded:: 0.20.0\n',
                   other='melt'))
    def melt(self, id_vars=None, value_vars=None, var_name=None,
             value_name='value', col_level=None):
        from pandas.core.reshape.melt import melt
        return melt(self, id_vars=id_vars, value_vars=value_vars,
                    var_name=var_name, value_name=value_name,
                    col_level=col_level)
```
This leaves the `@Appender` (I think it's a decorator) 1 indentation level bellow the function's signature, which is not proper syntax and raises another indentation error.

To avoid this problem, the present solution removes the same number os whitespaces from each line, as if the function was declared at an otherwise empty file:
```
@Appender(_shared_docs['melt'] %
          dict(caller='df.melt(',
               versionadded='.. versionadded:: 0.20.0\n',
               other='melt'))
def melt(self, id_vars=None, value_vars=None, var_name=None,
         value_name='value', col_level=None):
    from pandas.core.reshape.melt import melt
    return melt(self, id_vars=id_vars, value_vars=value_vars,
                var_name=var_name, value_name=value_name,
                col_level=col_level)
```

That said, I'm open to suggestions if someone can think of a simpler way to solve the described problem.